docs/topics/class-based-views/advanced.txt

```diff
@@ -0,0 +1,694 @@
+==============================
+
+As with the last example, let's look at what we need to reuse:
+``get_context_data()`` methods from both classes, and the ``post()`` method
+from ``FormView``.
+
+Unlike the last example, there is no clash between ``get()`` methods:
+``FormView``'s method does not need to do any special setup, therefore we can
+simply skip it by putting ``SingleObjectMixin`` first in the order of inheritance.
+
+As such, we don't need to override any methods at all! However, we may
+sometimes wish to override ``form_valid()`` to do some processing of the form
+(but would have to do the same in a stand-alone ``FormView``):
+
+.. snippet::
+    :filename: views.py
+
+    from django.views.generic import FormView
+    from django.shortcuts import redirect
+    from django import forms
+
+    class AuthorInterestForm(forms.Form):
+        message = forms.CharField()
+
+    class AuthorDetail2(SingleObjectMixin, FormView):
+        model = Author
+        form_class = AuthorInterestForm
+        template_name = "author2.html"
+
+        def form_valid(self, form):
+            # Here, we would have some logic based on user's form inputs
+            return redirect('author2', pk=self.get_object().pk)
+
+The template can use the ``author`` context variable and display the form:
+
+.. code-block:: html+django
+
+    <h2>Author {{ author.name }}</h2>
+
+    <form action="" method="post" accept-charset="utf-8">{% csrf_token %}
+        <table border="0" cellpadding="2" cellspacing="2">
+            {{ form.as_table }}
+        </table>
+    <p><input type="submit" value="Submit" /></p>
+    </form>
+
+Combined List View with a Form
+==============================
+
+This combined view is very similar to the last example. We need
+``MultipleObjectMixin`` to get the listing object into context and we need
+``FormView`` for ``post()`` method logic and adding the form to context. Since
+``FormView`` already provides a ``get()`` method, we don't need any additional
+classes:
+
+.. snippet::
+    :filename: views.py
+
+    class BooksFormView(MultipleObjectMixin, FormView):
+        model = Book
+        form_class = AuthorInterestForm
+        paginate_by = 2
+        template_name = "books.html"
+
+        def form_valid(self, form):
+            # Here, we would have some logic based on user's form inputs
+            return redirect('books-form')
+
+The template is also straightforward (pagination links could also be added as
+in the first example):
+
+.. code-block:: html+django
+
+    <h2>Books</h2>
+
+    <ol>
+      {% for book in page_obj %}
+        <li>{{ book.name }}</li>
+      {% endfor %}
+    </ol>
+
+
+    <form action="" method="post" accept-charset="utf-8">{% csrf_token %}
+        <table border="0" cellpadding="2" cellspacing="2">
+            {{ form.as_table }}
+        </table>
+    <p><input type="submit" value="Submit" /></p>
+    </form>
+
+A View with Arbitrary Number of Forms
+=====================================
```

Comment:
This has a horrible API. I don't disagree that a multiple forms generic view would be useful, (see django-extra-views), but I don't think this is the way to implement it.

For example, this will use the same `get_form_kwargs()` for each form, which is unlikely what you want.